File: SDKs/smart-me.com/v1/python/src/sdk/models/operations/meterfolderinformation_post.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class MeterFolderInformationPostRequests:
    application_xml: bytes = field(metadata={'request': { 'media_type': 'application/xml' }})
    text_xml: bytes = field(metadata={'request': { 'media_type': 'text/xml' }})
    meter_folder_information_to_post: Optional[shared.MeterFolderInformationToPost] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    meter_folder_information_to_post1: Optional[shared.MeterFolderInformationToPost] = field(default=None, metadata={'request': { 'media_type': 'application/x-www-form-urlencoded' }})
    meter_folder_information_to_post2: Optional[shared.MeterFolderInformationToPost] = field(default=None, metadata={'request': { 'media_type': 'text/json' }})
    

@dataclass
class MeterFolderInformationPostRequest:
    request: MeterFolderInformationPostRequests = field()
    

@dataclass
class MeterFolderInformationPostResponse:
    content_type: str = field()
    status_code: int = field()
    

File: SDKs/smart-me.com/v1/typescript/src/sdk/models/operations/meterfolderinformationpost.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class MeterFolderInformationPostRequests extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/xml" })
  applicationXml: Uint8Array;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  meterFolderInformationToPost?: shared.MeterFolderInformationToPost;

  @SpeakeasyMetadata({ data: "request, media_type=application/x-www-form-urlencoded" })
  meterFolderInformationToPost1?: shared.MeterFolderInformationToPost;

  @SpeakeasyMetadata({ data: "request, media_type=text/json" })
  meterFolderInformationToPost2?: shared.MeterFolderInformationToPost;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  textXml: Uint8Array;
}


export class MeterFolderInformationPostRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  request: MeterFolderInformationPostRequests;
}


export class MeterFolderInformationPostResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
